### ISSUE TITLE
[Bug]: Saving a fig with a colorbar using a `TwoSlopeNorm` sometimes results in 'posx and posy should be finite values'

### ISSUE THREAD

**Issue by Cgadal** [matplotlib/matplotlib#22300]
### Bug summary

When saving a figure with a colorbar and a mappable using a `TwoSlopeNorm`, it sometimes results in `posx and posy should be finite values`. The saved figure then lacks of a label for the colorbar.

Note that:
- you need to specify a `vmax` smaller than or equat to `0.9989999999999999`
- this looks independent of the max and min values of `Z`. Tested with `Z = 10000*(X**3 - Y**4)` and `Z = (X**3 - Y**4)/10000`

it dos not happen:
- if the figure is not saved but only shown
- with previous versions of matplotlib

### Code for reproduction

```python
import matplotlib.pyplot as plt
import numpy as np
import matplotlib.colors as colors

x = np.linspace(0, 1, 100)
y = x.copy()
X, Y = np.meshgrid(x, y)
Z = X**3 - Y**4


divnorm = colors.TwoSlopeNorm(vcenter=0, vmax=0.1)

fig = plt.figure()
c = plt.pcolormesh(x, y, Z, cmap='seismic', norm=divnorm)
plt.colorbar(c, label='Velocity [cm/s]')
plt.savefig('Spatio_temporal_velocity.png')
plt.show()
```

### Actual outcome

![divnorm](https://user-images.githubusercontent.com/48522112/150763120-f2d65fa0-7b41-4ae4-9237-1f0747c49598.png)

### Expected outcome

It should display the colorbar label, and not return in the terminal `posx and posy should be finite values`.

### Additional information

- you need to specify a `vmax` smaller than or equat to `0.9989999999999999`
- this looks independent of the max and min values of `Z`. Tested with `Z = 10000*(X**3 - Y**4)` and `Z = (X**3 - Y**4)/10000`

it dos not happen:
- if the figure is not saved but only shown
- with previous versions of matplotlib

### Operating system

Ubuntu 20

### Matplotlib Version

3.5.0

### Matplotlib Backend

QtAgg

### Python version

3.8.10

### Installation

pip

**Comment by jklymak** [comment]
This is somehow in the `_update_label_position` code.  

Until we fix this, if you do `fig.draw_without_rendering()` before saving, then the label works fine.  Somehow the extent of the colorbar is not correct until we do that.

**Comment by jklymak** [comment]
Thanks for the clear report - should be fixed in 3.5.2

**Comment by a1324664328** [comment]
I went through the code line by line and at first thought it was a data out of bounds. Man, what can I say?
